core-bundle/src/Resources/contao/dca/tl_templates.php

```diff
@@ -499,9 +502,10 @@ public function compareTemplate(DataContainer $dc)
 		$objTemplate->base = Environment::get('base');
 		$objTemplate->language = $GLOBALS['TL_LANGUAGE'];
 		$objTemplate->title = StringUtil::specialchars($GLOBALS['TL_LANG']['MSC']['showDifferences']);
-		$objTemplate->charset = Config::get('characterSet');
+		$objTemplate->charset = System::getContainer()->getParameter('kernel.charset');
 
-		Config::set('debugMode', false);
+		System::getContainer()->setParameter('kernel.debug', false);
+		$GLOBALS['TL_CONFIG']['debugMode'] = false;
```

Comment:
Using `$GLOBALS` is worse than using `Config:set()`, isn't it?

Reply:
Yes but I wanted to keep it for BC. Our own code uses the `kernel.debug` parameter and doesn’t need the global, but third party code still might use that.